File: Space_boot/src/main/java/space/model/Possession.java
package space.model;


import jakarta.persistence.*;

@Entity
@Table(name = "possession")
public class Possession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private int quantite;
    @Enumerated(EnumType.STRING)
    @Column(name = "ressource", columnDefinition = "ENUM('ARME','NOURRITURE','ENERGIE','ARGENT')", nullable = false)
    private Ressource ressource;


    public Possession() {
    }

    public Possession(Integer id, int quantite, Ressource ressource) {
        this.id = id;
        this.quantite = quantite;
        this.ressource = ressource;
    }

    public Possession(int quantite, Ressource ressource) {
        this.quantite = quantite;
        this.ressource = ressource;
    }


    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public Ressource getRessource() {
        return ressource;
    }

    public void setRessource(Ressource ressource) {
        this.ressource = ressource;
    }

    @Override
    public String toString() {
        return "Possession [id=" + id + ", quantite=" + quantite + ", ressource=" + ressource + "]";
    }


}

File: Space_front/src/app/possession.ts
export class Possession {
  constructor(
    private _id: number,
    private _quantite: number,
    private _ressource: string
  ) {}

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get quantite(): number {
    return this._quantite;
  }

  set quantite(value: number) {
    this._quantite = value;
  }

  get ressource(): string {
    return this._ressource;
  }

  set ressource(value: string) {
    this._ressource = value;
  }
}
